api/src/test/java/org/openmrs/validator/VisitValidatorTest.java

```diff
@@ -448,4 +448,52 @@ public void validate_shouldFailValidationIfFieldLengthsAreNotCorrect() {
 		new VisitValidator().validate(visit, errors);
 		assertEquals(true, errors.hasFieldErrors("voidReason"));
 	}
+	
+	/**
+	 * @see VisitValidator#validate(Object,Errors)
+	 */
+	@Test
+	public void validate_shouldFailValidationIfVisitDateBeforeBirthDate() {
+		VisitService vs = Context.getVisitService();
+		Visit visit = new Visit();
+		
+		visit.setVisitType(vs.getVisitType(1));
+		visit.setPatient(Context.getPatientService().getPatient(2));
+		Calendar c = Calendar.getInstance();
+		c.set(1974, 4, 8);
+		visit.setStartDatetime(c.getTime());
+		
+		Errors errors = new BindException(visit, "visit");
+		new VisitValidator().validate(visit, errors);
+		assertTrue(errors.hasFieldErrors("startDatetime"));
```

Comment:
can you assert that the case you expect is actually occuring by using an assertion we built `hasFieldErrors` you can see an example here:

https://github.com/openmrs/openmrs-core/blob/master/api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java#L89-L99

also this test example shows the AAA style and that there is the least clutter necessary to see whats tested. Maybe you can apply some more cleanup to this test class according to the example test.